ref(sentry apps): Reorganize UI components info (#4994)
getsentry/sentry-docs: src/docs/product/integrations/integration-platform/ui-components/issue-link.mdx

![Issue linking sidebar showing Shortcut icon.](issue-sidebar-icons.png)

Clicking the link opens a modal allowing the user to create or link the issue to a task in the external service:
![Sentry modal that creates a Shortcut Story that's linked to a Sentry issue.](../../issue-tracking/shortcut/shortcut_create_story_with_issue.png)

When an external issue is created or linked in [sentry.io](https://sentry.io), a display name is shown that links back to the service where it was either created or linked. The display name is composed of two pieces: the `project` and the `identifier`. A hash (#) connects each piece. Here's an example of what it looks like in [sentry.io](https://sentry.io):

![Sentry's sidebar button illustrating that an issue is linked.](../../issue-tracking/shortcut/shortcut_story_in_linked_issues_82.png)

## Schema

```json {filename:schema.json}
{
  "elements": [
    {
      "type": "issue-link",
      "link": {
        "uri": <String>,
        "required_fields": <Array<FormField>>,
        "optional_fields": <Array<FormField>>
      },
      "create": {
        // Same as "link" schema
      }
    }
  ]
}
```

> ### Attributes

Reviewer: This can be an h2 to match the other UI component pages, but I also get if you want to keep it h3, since there is already another 'Attributes' heading on the page that makes sense to be a h3

---
title: Issue Link
sidebar_order: 1
description: "Learn about the issue link UI component."
---

With an issue linking UI component, the developer can connect Sentry issues with a task in their project management system. This functionality provides a way to use any project management tool the developer uses or develops.

The Issue Link component displays "Link &lt;Service&gt; Issue" in the Issue sidebar, which opens a modal allowing the user to create or link the issue to a task in the external service. For example, this is what our Shortcut integration looks like.

![Issue linking sidebar showing Shortcut icon.](issue-sidebar-icons.png)

> 

Author: Since the text above describes what it is and where to find it, I'm moving the image just above it. Lmk if that doesn't make sense.

> 

Reviewer 2: What about breaking it up like so: 

The issue link component displays the "Link<Service> Issue" in the Issue Details sidebar. 
image 1

Clicking this opens...
image 2

![Sentry modal that creates a Shortcut Story that's linked to a Sentry issue.](../../issue-tracking/shortcut/shortcut_create_story_with_issue.png)

## Issue Link Schema

```json
{
  "type": "issue-link",
  "link": {
    "uri": <String>,
    "required_fields": <Array<FormField>>,
    "optional_fields": <Array<FormField>>
  },
  "create": {
    // Same as "link" schema
  }
}
```

> ### Attributes of the link and create fields

Reviewer 2: ```suggestion
### Attributes of the Link and Create Fields
```

- `optional_fields` - (Optional) List of [FormField](/product/integrations/integration-platform/ui-components/formfield) components the User may complete.

## Issue Link Request and Response

When a user attempts to create or link an issue, we will send a request to your service based on the provided `uri`.

### Request Format

We send a `POST` request, so all the information is stored in the body.

### Schema

```json
{
  "fields": <Object>,
  "installationId": <String>,
  "issueId": <String>,
  "webUrl": <String>,
  "project": {"slug": <String>, "id": <String>},
  "actor": {"type": "user", "name": <String>, "id": <String>},
}
```

### Attributes

- `fields` - Key/value pairs of the data in the form fields
- `installationId` - The ID of the installation associated with the request
- `issueId` - The ID of the issue in Sentry
- `webUrl` - The URL of the issue in Sentry
- `project`
  - `slug` - The slug of the project in Sentry
  - `id` - The ID of the project in Sentry
- `actor`
  - `name` - The name of the user in Sentry
  - `id` - The ID of user in Sentry

### Response Format

> When creating or linking an issue, the response format _must_ have the following schema:
> 
> ### Schema
> 

Reviewer 2: ```suggestion
### Schema

When creating or linking an issue, the response format _must_ have the following schema:

```

```json
{
  "webUrl": <String>,
  "project": <String>,
  "identifier": <String>,
}
```

### Attributes

- `webUrl`- The URL of the linked issue in your project management system
- `project`- The first part of the displayed issue that should probably be associated with a project in your system
- `identifier`- The second part of the displayed issue that should be unique to the project

> When an external issue is created or linked in Sentry, Sentry shows a display name that links back to the service where it was either created or linked. The display name is composed of two pieces: the `project` and the `identifier`. A hash (#) connects each piece. Here's an example of what it looks like in Sentry:

Reviewer 2: ```suggestion
When an external issue is created or linked in [sentry.io](https://sentry.io), a display name is shown that links back to the service where it was either created or linked. The display name is composed of two pieces: the `project` and the `identifier`. A hash (#) connects each piece. Here's an example of what it looks like in [sentry.io](https://sentry.io):
```

---
title: Issue Link
sidebar_order: 1
description: "Learn about the issue link UI component."
---

With an issue linking UI component, you can connect Sentry issues with a task in your project management system. This functionality provides a way to use any project management tool you use or develop.

> The issue link component displays with the text "Link &lt;Service&gt; Issue" in the **Issue Details** sidebar.

Reviewer 2: ```suggestion
The issue link component displays with the text "Link &lt;Service&gt; Issue" in the **Issue Details** sidebar:
```

![Issue linking sidebar showing Shortcut icon.](issue-sidebar-icons.png)

> Clicking the link opens a modal allowing the user to create or link the issue to a task in the external service.

Reviewer 2: ```suggestion
Clicking the link opens a modal allowing the user to create or link the issue to a task in the external service:
```

![Sentry modal that creates a Shortcut Story that's linked to a Sentry issue.](../../issue-tracking/shortcut/shortcut_create_story_with_issue.png)

> ## Issue Link Schema

Reviewer: To be consistent with the other UI Component pages, I think think this header should be changed to just `Schema` (or you can change the others to this pattern)

```json {filename:schema.json}
{
  "elements": [
    {
      "type": "issue-link",
      "link": {
        "uri": <String>,
        "required_fields": <Array<FormField>>,
        "optional_fields": <Array<FormField>>
      },
      "create": {
        // Same as "link" schema
      }
    }
  ]
}
```

> ### Attributes of the Link and Create Fields

Reviewer: Again, to go with the pattern on the other pages, I suggest changing this to just `Attributes`

- `uri` - (Required) The URI to request when the User submits the Link/Create Issue form.
- `required_fields` - (Required) List of [FormField](/product/integrations/integration-platform/ui-components/formfield) components the User is required to complete.
- `optional_fields` - (Optional) List of [FormField](/product/integrations/integration-platform/ui-components/formfield) components the User may complete.

## Issue Link Request and Response

When a user attempts to create or link an issue, we will send a request to your service based on the provided `uri`.

### Request Format

> We send a `POST` request, so all the information is stored in the body.

Reviewer: I think these two sentences can be written one after another and a heading removed so that we just have `## Request Format` as an h2. I'm just judging by the sidebar, and it feels better like that.

### Schema

> ```json {filename:schema.json}

Reviewer: I personally think this heading should just be removed, since the below isn't a schema either, it's a response format.

You should remove the filename and get rid of the `"elements"` key entirely.

Author: lol woops, got caught in my big sweep

{
  "elements": [
    {
      "fields": <Object>,
      "installationId": <String>,
      "issueId": <String>,
      "webUrl": <String>,
      "project": {"slug": <String>, "id": <String>},
      "actor": {"type": "user", "name": <String>, "id": <String>},
    }
  ]
}
```

### Attributes

- `fields` - Key/value pairs of the data in the form fields
- `installationId` - The ID of the installation associated with the request
- `issueId` - The ID of the issue in Sentry
- `webUrl` - The URL of the issue in Sentry
- `project`
  - `slug` - The slug of the project in Sentry
  - `id` - The ID of the project in Sentry
- `actor`
  - `name` - The name of the user in Sentry
  - `id` - The ID of user in Sentry

### Response Format

> ### Schema

Reviewer: Again, I feel this looks weird on the sidebar, so I vote to remove `Schema` as a heading (since it isn't one) and change Response Format to an h2 `## Response Format`

When creating or linking an issue, the response format _must_ have the following schema:

```json
{
  "webUrl": <String>,
  "project": <String>,
  "identifier": <String>,
}
```

### Attributes

- `webUrl`- The URL of the linked issue in your project management system
- `project`- The first part of the displayed issue that should probably be associated with a project in your system
- `identifier`- The second part of the displayed issue that should be unique to the project

When an external issue is created or linked in [sentry.io](https://sentry.io), a display name is shown that links back to the service where it was either created or linked. The display name is composed of two pieces: the `project` and the `identifier`. A hash (#) connects each piece. Here's an example of what it looks like in [sentry.io](https://sentry.io):

> ![Sentry's sidebar button illustrating that an issue is linked.](../../issue-tracking/shortcut/shortcut_story_in_linked_issues_82.png)

Reviewer: This section of info should be located just after the modal form image, not under this heading.

## Example

> ```json

Reviewer: Filename for this one as well! And to match the other pages, I think this should be moved to just after the `Attributes` heading.
